### PR TITLE
Add redux to components

### PR DESCRIPTION
**Pull request by deyemiobaa**
### Add redux to components

- Keep state immutable in reducers
- Use `useSelector` to consume the state.
- Dispatch a corresponding action for adding and removing books
- Dispatch a corresponding action for categories

### CODE REVIEW

File: src/components/form/book-submit.js [hunk 1]
```diff
@@ -1,34 +1,53 @@
...
-          </button>
-        </form>
-      </>
-    );
-  }
-}
+const Form = () => {
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
```
Comment (EricMbouwe) [inline src/components/form/book-submit.js]: - [ ] This component should have a local state with  `title`, `author`
- [ ] You should have your title and author inputs that update the local state every time they update with an (`onChange`) method

File: src/components/form/book-submit.js [hunk 2]
```diff
@@ -1,34 +1,53 @@
...
+      <h2>Add New Book</h2>
+      <form onSubmit={handleSubmit}>
+        <div className="form-group">
+          <input
+            type="text"
+            className="form-control"
+            name="title"
+            placeholder="Title"
+            required
+          />
```
Comment (EricMbouwe) [inline src/components/form/book-submit.js]: - [ ] Let's add an onChange event prop here to handle the input update

File: src/components/form/book-submit.js [hunk 3]
```diff
@@ -1,34 +1,53 @@
...
+        </div>
+        <div className="form-group">
+          <input
+            type="text"
+            className="form-control"
+            name="category"
+            placeholder="Category"
+            required
+          />
```
Comment (EricMbouwe) [inline src/components/form/book-submit.js]: - [ ] Let's add an onChange event prop here to handle the input update

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @deyemiobaa,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @deyemiobaa,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**